core/src/main/java/com/scalar/db/transaction/consensuscommit/CommitHandler.java

```diff
@@ -170,6 +187,52 @@ public void commit(Snapshot snapshot, boolean readOnly)
     }
   }
 
+  @VisibleForTesting
+  boolean canOnePhaseCommit(Snapshot snapshot) throws CommitException {
+    if (!onePhaseCommitEnabled) {
+      return false;
+    }
+
+    // If validation is required (in SERIALIZABLE isolation), we cannot one-phase commit the
+    // transaction
+    if (snapshot.isValidationRequired()) {
+      return false;
+    }
+
+    // If the snapshot has no write and deletes, we do not one-phase commit the transaction
+    if (!snapshot.hasWritesOrDeletes()) {
+      return false;
+    }
+
+    List<Delete> deletesInDeleteSet = snapshot.getDeletesInDeleteSet();
+
+    // If a record corresponding to a delete in the delete set does not exist in the storage,　we
+    // cannot one-phase commit the transaction. This is because the storage does not support
+    // delete-if-not-exists semantics, so we cannot detect conflicts with other transactions.
+    for (Delete delete : deletesInDeleteSet) {
```

Comment:
I wanted to know what would happen if I removed this block, so I removed the block and executed `ConsensusCommitNullMetadataIntegrationTestBase`. But all the tests passed unexpectedly. Is it difficult to reproduce a case where this block is necessary? (or I'm missing something....)

Reply:
I see. Thanks for the explanation! It would be great if we add a test case for the situation later.